include/envoy/network/transport_socket.h

```diff
@@ -151,9 +142,9 @@ class TransportSocket {
   virtual void onConnected() PURE;
 
   /**
-   * @return the const SSL connection data if this is an SSL connection, or nullptr if it is not.
+   * @return the SSL connection data if this is an SSL connection, or nullptr if it is not.
    */
-  virtual Ssl::ConnectionInfoConstSharedPtr ssl() const PURE;
+  virtual Ssl::ConnectionInfoSharedPtr ssl() const PURE;
```

Comment:
Do we need to change all ConstSharedPtr to SharedPtr? For usecases other than handshake it should remain const?

Reply:
I have changed it back to ConnectionInfoConstSharedPtr in all cases except for the Network::Connection and ::TransportSocket, which (I think) both need to expose a non-const Ssl::ConnectionInfoSharedPtr ssl() in order to make this work.

Do you think it's worth having two ssl() accessor methods, one const and one non-const? Not sure how good style that is.

Reply:
I'm not following, why do we need this interface to return a non ConstSharedPtr? I think everywhere you need a non-const shared_ptr is in the implementation? `info_->doHandshake();` is the only place, but you can just make the member_ is non-const, but return a ConstSharedPtr.

Reply:
I don't know why I thought I needed a non-const shared ptr. I've reverted this change and can now pass around a ConstSharedPtr everywhere. Sorry for the confusion.

Reply:
I think you needed when you had `doHandshake` in ConnectionInfo, and now it's no longer the case :).